### ISSUE TITLE
[PATCH] fixed usage of "no if .. warnings" in perldelta

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13006]
Migrated from [rt.perl.org#118249](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=118249) (status was 'resolved')

Searchable as RT118249$

**Comment by p5pRT** [comment]
### From @domm


This is a bug report for perl from domm@&#8203;plix\.at\,
generated with the help of perlbug 1\.39 running under perl 5\.14\.2\.

From eeeadd16dcfaa7977dfba9a1ba07ac7708ad16f0 Mon Sep 17 00&#8203;:00&#8203;:00 2001
From&#8203;: Thomas Klausner \<domm@&#8203;plix\.at>
Date&#8203;: Thu\, 30 May 2013 11&#8203;:10&#8203;:30 \+0200
Subject&#8203;: \[PATCH\] fixed usage of "no if \.\. warnings" in perldelta



---
 pod/perl5180delta\.pod |    2 \+\-
 1 file changed\, 1 insertion\(\+\)\, 1 deletion\(\-\)

<details><summary>Inline Patch</summary>

```diff
diff --git a/pod/perl5180delta.pod b/pod/perl5180delta.pod
index aa244dc..03341b6 100644
--- a/pod/perl5180delta.pod
+++ b/pod/perl5180delta.pod
@@ -405,7 +405,7 @@ is not recommended.
 Warnings will now be issued when the parser sees C<~~>, C<given>, or C<when>.
 To disable these warnings, you can add this line to the appropriate scope:
 
-  no if $] >= 5.018, "experimental::smartmatch";
+  no if $] >= 5.018, 'warnings', "experimental::smartmatch";
 
 Consider, though, replacing the use of these features, as they may change
 behavior again before becoming stable.
-- 

```
</details>
1.7.10.4



---
Flags&#8203;:
&nbsp;    category=docs
&nbsp;    severity=low


---
Site configuration information for perl 5\.14\.2&#8203;:

Configured by Debian Project at Wed Apr 10 16&#8203;:58&#8203;:16 UTC 2013\.

Summary of my perl5 \(revision 5 version 14 subversion 2\) configuration&#8203;:
&nbsp;   
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.2\.0\-4\-amd64\, archname=x86\_64\-linux\-gnu\-thread\-multi
&nbsp;    uname='linux madeleine 3\.2\.0\-4\-amd64 \#1 smp debian 3\.2\.39\-2 x86\_64 gnulinux '
&nbsp;    config\_args='\-Dusethreads \-Duselargefiles \-Dccflags=\-DDEBIAN \-D\_FORTIFY\_SOURCE=2 \-g \-O2 \-fstack\-protector \-\-param=ssp\-buffer\-size=4 \-Wformat \-Werror=format\-security \-Dldflags= \-Wl\,\-z\,relro \-Dlddlflags=\-shared \-Wl\,\-z\,relro \-Dcccdlflags=\-fPIC \-Darchname=x86\_64\-linux\-gnu \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.14 \-Darchlib=/usr/lib/perl/5\.14 \-Dvendorprefix=/usr \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/perl5 \-Dsiteprefix=/usr/local \-Dsitelib=/usr/local/share/perl/5\.14\.2 \-Dsitearch=/usr/local/lib/perl/5\.14\.2 \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3 \-Dsiteman1dir=/usr/local/man/man1 \-Dsiteman3dir=/usr/local/man/man3 \-Duse64bitint \-Dman1ext=1 \-Dman3ext=3perl \-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Ud\_ualarm \-Uusesfio \-Uusenm \-Ui\_libutil \-DDEBUGGING=\-g \-Doptimize=\-O2 \-Duseshrplib \-Dlibperl=libperl\.so\.5\.14\.2 \-des'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fstack\-protector \-fno\-strict\-aliasing \-pipe \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O2 \-g'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fstack\-protector \-fno\-strict\-aliasing \-pipe \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.7\.2'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib /usr/lib
&nbsp;    libs=\-lgdbm \-lgdbm\_compat \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    libc=\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.14\.2
&nbsp;    gnulibc\_version='2\.13'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    



---
@&#8203;INC for perl 5\.14\.2&#8203;:
&nbsp;    /etc/perl
&nbsp;    /usr/local/lib/perl/5\.14\.2
&nbsp;    /usr/local/share/perl/5\.14\.2
&nbsp;    /usr/lib/perl5
&nbsp;    /usr/share/perl5
&nbsp;    /usr/lib/perl/5\.14
&nbsp;    /usr/share/perl/5\.14
&nbsp;    /usr/local/lib/site\_perl
&nbsp;    \.



---
Environment for perl 5\.14\.2&#8203;:
&nbsp;    HOME=/home/domm
&nbsp;    LANG \(unset\)
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LC\_CTYPE=en\_US\.UTF\-8
&nbsp;    LC\_MESSAGES=POSIX
&nbsp;    LC\_MONETARY=de\_AT\.utf8
&nbsp;    LC\_NUMERIC=de\_AT\.utf8
&nbsp;    LC\_PAPER=de\_AT\.utf8
&nbsp;    LC\_RESPONSE=POSIX
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/home/domm/perl5/perlbrew/bin&#8203;:/home/domm/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/local/games&#8203;:/usr/games&#8203;:/home/domm/bin
&nbsp;    PERLBREW\_BASHRC\_VERSION=0\.63
&nbsp;    PERLBREW\_HOME=/home/domm/\.perlbrew
&nbsp;    PERLBREW\_ROOT=/home/domm/perl5/perlbrew
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @karenetheridge


On Thu\, May 30\, 2013 at 02&#8203;:20&#8203;:45AM \-0700\, Thomas Klausner wrote&#8203;:
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=118249 >
>
> \-  no if $\] >= 5\.018\, "experimental&#8203;::smartmatch";
> \+  no if $\] >= 5\.018\, 'warnings'\, "experimental&#8203;::smartmatch";

I'd change that to&#8203;:

&nbsp;   no if $\] >= 5\.018\, warnings => "experimental&#8203;::smartmatch";

\.\.\.to make it more clear that it's the warnings pragma being called\,
with "experimental&#8203;::smartmatch" as its argument\, rather than two unrelated
arguments being used\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @Hugmeir

On Thu\, May 30\, 2013 at 6&#8203;:17 PM\, Karen Etheridge \<perl@&#8203;froods\.org> wrote&#8203;:

>
> On Thu\, May 30\, 2013 at 02&#8203;:20&#8203;:45AM \-0700\, Thomas Klausner wrote&#8203;:
> > \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=118249 >
> >
> > \-  no if $\] >= 5\.018\, "experimental&#8203;::smartmatch";
> > \+  no if $\] >= 5\.018\, 'warnings'\, "experimental&#8203;::smartmatch";
>
> I'd change that to&#8203;:
>
>    no if $\] >= 5\.018\, warnings => "experimental&#8203;::smartmatch";
>
> \.\.\.to make it more clear that it's the warnings pragma being called\,
> with "experimental&#8203;::smartmatch" as its argument\, rather than two unrelated
> arguments being used\.
>
>
\+1

**Comment by p5pRT** [comment]
### From @rjbs

How embarrassing\!

I have fixed this in blead as 2153ce5 and maint\-5\.18 as 576348d\, using the tweak that Karen 
suggested\.  Thanks\.
 
\-\- 
rjbs

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

How embarrassing\!

I have fixed this in blead as 2153ce5 and maint\-5\.18 as 576348d\, using the tweak that Karen 
suggested\.  Thanks\.
 
\-\- 
rjbs

**Comment by p5pRT** [comment]
@rjbs - Status changed from 'open' to 'resolved'
